test/testother.cpp

```diff
@@ -7583,6 +7583,20 @@ class TestOther : public TestFixture {
               "    }\n"
               "};\n", nullptr, /*experimental*/ false, /*inconclusive*/ true);
         ASSERT_EQUALS("", errout.str());
+
+        // #10704
+        check("struct C {\n"
+              "    std::string str;\n"
+              "    const std::string& get() const { return str; }\n"
+              "};\n"
+              "struct D {\n"
+              "    C c;\n"
+              "    bool f() const {\n"
+              "        std::string s = c.get();\n"
+              "        return s.empty();\n"
+              "    }\n"
+              "};\n");
+        ASSERT_EQUALS("[test.cpp:8]: (performance, inconclusive) Use const reference for 's' to avoid unnecessary data copying.\n", errout.str());
```

Comment:
could you add a test case here to ensure that you don't warn when the parent object `c` is temporary instead.

maybe something like this for instance:

    C getC();

    void f() {
        std::string s = getC().get();
    }

Reply:
Current head warns for this code as soon as `s` is `const`, which, as mentioned above, is not a good heuristic. The idea was probably that someone just forgot to add `&`.
~~~
struct C {
	const std::string& get() const;
};
C getC();

void f() {
    const std::string s = getC().get();
}
~~~

Reply:
oops.. so that should be fixed